modules/nlu/src/backend/language/language-provider.ts

```diff
@@ -135,6 +150,56 @@ export class RemoteLanguageProvider implements LanguageProvider {
     return this
   }
 
+  private extractLangServerVersion(data) {
+    const version = semver.valid(semver.coerce(data.version))
+
+    if (!version) {
+      throw new Error('Lang server has an invalid version')
+    }
+    this._state.langServerVersion = semver.clean(version)
```

Comment:
Sorry I think I wasn't clear

we need to clean the cache on the when the language server info changed i.e 
dimension, domain, maybe version as well

Reply:
Ok sure embedding dimensions are an important thing to consider for the hash, but what about domain?

If the same version of the lang server runs on a different domain, do we really want to clear cache?